src/dp_transformers/grad_sample/transformers/conv_1d.py

```diff
@@ -1,27 +1,22 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
-import torch
-
-from transformers.modeling_utils import Conv1D
+from typing import Dict
 
+import torch
+import torch.nn as nn
 from opacus.grad_sample import utils
+from transformers.modeling_utils import Conv1D
 
 
 def compute_transformers_conv1d_grad_sample(
-    layer: Conv1D, A: torch.Tensor, B: torch.Tensor, batch_dim: int = 0
-) -> None:
-    gs = torch.einsum("n...i,n...j->nji", B, A).contiguous()
-    utils.create_or_extend_grad_sample(
-        layer.weight, gs, batch_dim
-    )
+    layer: Conv1D, A: torch.Tensor, B: torch.Tensor
+) -> Dict[nn.Parameter, torch.Tensor]:
+    ret = {}
+    ret[layer.weight] = torch.einsum("n...i,n...j->nji", B, A).contiguous()
     if layer.bias is not None:
-        utils.create_or_extend_grad_sample(
-            layer.bias,
-            torch.einsum("n...k->nk", B),
-            batch_dim,  # pyre-ignore[6] We know layer.bias is not None
-        )
-
+        ret[layer.bias] = torch.einsum("n...k->nk", B)
+    return ret
 
 def register_grad_sampler() -> None:

```

Comment:
Can we remove this and do it like the new version does with the decorators? https://github.com/pytorch/opacus/blob/main/opacus/grad_sample/embedding.py

I think then we'd just need to import the file instead of calling this function